pinc/misc.inc

```diff
@@ -130,7 +130,6 @@ function get_enumerated_param($arr, $key, $default, $choices, $allownull = false
     {
         // sanity checks on the args
         assert(is_array($arr));
-        assert(!is_array($key));
```

Comment:
I've landed on only being comfortable removing these checks from our param validation functions if we add type declarations to the function calls such that if an invalid type is passed in (almost always developer error) we will get a very loud exception rather than a silent failure. In fact, if we add PHP type declarations to these we can probably remove most of these assert checks.

If we do that, we should do it in a separate PR to all `get_*_param()` functions (including `get_round_param()` in `Round.inc`) and probably want a sandbox to sanity check. We could also add unit tests to these in `SETUP/tests/misc_ParamValidatorTest.php`.

Reply:
Requiring type declarations makes sense. I assume you mean either explicitly typed PHPDocs `@param`s and `@return`s (which PHPStan already uses for type inference), or PHP typehints on method/function signatures (which PHPStan can also read).

This particular function already has `@params` which is how PHPStan was able to tell us some of the asserts were redundant. Of course for ahead of time static type checking to be an effective guard, we need to use it for all the callers, and actually pay attention to the warnings.

Reply:
I mean [PHP type declarations](https://www.php.net/manual/en/language.types.declarations.php) because PHP will raise an exception if what passes in doesn't match the signature.

Reply:
I can work on getting those done if you haven't started down that path already.

Reply:
OK. I haven't started yet. Whichever of us gets to it first! :)

Reply:
See https://github.com/DistributedProofreaders/dproofreaders/pull/1079